synapse/rest/client/v2_alpha/sendtodevice.py

```diff
@@ -42,7 +43,10 @@ def __init__(self, hs):
         self.txns = HttpTransactionCache(hs)
         self.device_message_handler = hs.get_device_message_handler()
 
+    @trace_using_operation_name("sendToDevice")
```

Comment:
suggestion: instead of having a whole separate `trace_using_operation_name`, can we just make `trace` have an optional `opname` param?

Reply:
then we'd have to write @trace() each time. I'd have to do this in a separate pr because this will affect everything

Reply:
> then we'd have to write @trace() each time

Alternatively: make the `func` param optional on `trace`. If it is unset, then it was invoked as

```
@trace()
def foo(): 
    # ...
```

and it should return a decorator function. If it is set, then it should do the decoration itself and return the decorated function.

Reply:
Oooh, clever. I like that